### PR TITLE
Refactor outcome creation

### PR DESCRIPTION
**Pull request by derekprior**
* Use accepts_nested_attributes to handle the alignment creation rather
  than handling it manually.
* Perform the multiple creation steps in a transaction to ensure
  integrity if one step fails.
* Use Simple Form to build the forms with less markup

### CODE REVIEW

File: app/controllers/outcomes_controller.rb [hunk 1]
```diff
@@ -7,37 +7,48 @@ def show
...
+  def course
+    @course ||= Course.find(params[:course_id])
+  end
+
+  def create_outcome(outcome)
+    ActiveRecord::Base.transaction do
+      authorize(outcome)
+      outcome.course.adopt_custom_outcomes!
+      outcome.save!
+      true
```
Comment (mrgilman) [inline app/controllers/outcomes_controller.rb]: Do you need this line if `outcome.save!` returns `true`?

**Comment by mrgilman** [comment]
:volcano:

**Comment by mrgilman** [comment]
(That's a good thing)

File: app/controllers/outcomes_controller.rb (hunk 1 above)
Comment (derekprior) [inline app/controllers/outcomes_controller.rb]: Good call.

**Pull request merged**